extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/RuntimeBeanProcessor.java

```diff
@@ -0,0 +1,66 @@
+package io.quarkus.arc.deployment;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+
+import io.quarkus.arc.runtime.ArcDeploymentTemplate;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.gizmo.ClassCreator;
+import io.quarkus.gizmo.ClassOutput;
+import io.quarkus.gizmo.FieldDescriptor;
+import io.quarkus.gizmo.MethodCreator;
+import io.quarkus.gizmo.MethodDescriptor;
+import io.quarkus.gizmo.ResultHandle;
+
+public class RuntimeBeanProcessor {
+
+    @BuildStep
+    @Record(ExecutionTime.STATIC_INIT)
+    void build(List<RuntimeBeanBuildItem> beans,
+            BuildProducer<GeneratedBeanBuildItem> generatedBean,
+            ArcDeploymentTemplate template) {
+        int count = 0;
+        Map<String, Supplier<Object>> map = new HashMap<>();
+        for (RuntimeBeanBuildItem b : beans) {
+            String name = "rb" + count;
+            map.put(name, b.supplier);
+            String beanName = "io.quarkus.arc.runtimebean.GeneratedBean" + count;
```

Comment:
Hm, this is not deterministic. Why don't we use a single `GeneratedBean` class for all producer methods? Where a producer method name is derived from the `GeneratedBeanBuildItem`?

Reply:
Good idea